editor/effects.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+import { parse, stringify } from 'querystring';
+
+/**
+ * WordPress dependencies
+ */
+import { getBlockSettings, switchToBlockType } from 'blocks';
+
+/**
+ * Internal dependencies
+ */
+import { focusBlock, replaceBlocks } from './actions';
+
+export default {
+	REQUEST_POST_UPDATE( action, store ) {
+		const { dispatch } = store;
+		const { postId, edits } = action;
+		const isNew = ! postId;
+		const toSend = postId ? { id: postId, ...edits } : edits;
+
+		new wp.api.models.Post( toSend ).save().done( ( newPost ) => {
+			dispatch( {
+				type: 'REQUEST_POST_UPDATE_SUCCESS',
+				post: newPost,
+			} );
+		} ).fail( ( err ) => {
+			dispatch( {
+				type: 'REQUEST_POST_UPDATE_FAILURE',
+				error: get( err, 'responseJSON', {
```

Comment:
>Minor:  I think it would be nicer to extract both, `REQUEST_POST_UPDATE_SUCCESS` and `REQUEST_POST_UPDATE_FAILURE` into an action creator, obviously not a high priority and maybe can be done if/when these actions are triggered elsewhere, to ensure consistency.

I've been thinking on success and failure types generally, and would like to revisit them in a subsequent pull request. Historically the triple of `REQUEST`, `_SUCCESS`, and `_FAILURE` types has been a pain point for me and I don't think they're always necessary. I've talked with a few people about what it means to eliminate them in favor of (a) more optimistic state changes and (b) more semantic "received updated version" actions. For example, [something like this](https://github.com/WordPress/gutenberg/blob/91c208e8eab031d0cd58dc7ca2b38428c85bb084/editor/state.js#L174-L175) should ideally not be tied so strongly to the success case, but rather the success case dispatching an action indicating that the latest version of the post has been received. Where success and failure types could be necessary is in indicating progress, such as the "Saving..." text to be shown on the publish button, or in presenting errors in the UI.